### ISSUE TITLE
BloomEffect stops working with Three@153

### ISSUE THREAD

**Issue by lviggiani** [pmndrs/postprocessing#501]
I'm using `BloomEffect` on emissive materials with success as you taught here:

https://github.com/pmndrs/postprocessing/issues/496

It works fine with Three `150`, `151` and `152`

![image](https://github.com/pmndrs/postprocessing/assets/139040/44c27142-3146-4c05-a860-a5f3e1027d8e)

But when I switch to Three `153` bloom doesn't work anymore

![image](https://github.com/pmndrs/postprocessing/assets/139040/6d342ab4-ee61-4a61-80e7-1581927744f9)

I'm using latest `6.31.2` version of `postprocessing.esm.js` in all the above cases and exaclty same code on my app

**Comment by vanruesc** [comment]
Strange, going from three r152 to 153 shouldn't break this. Are you sure nothing else has changed during the update? It looks like the headlight emissive color isn't as bright as before.

**Comment by CodyJasonBennett** [comment]
Likely related to https://github.com/mrdoob/three.js/pull/26130 which clamps values after tonemapping. Make sure to set `Material.toneMapped = false` when working with HDR effects.

**Comment by CodyJasonBennett** [comment]
The code there looks fine, but you'd need a tonemapping pass afterward if you do want realistic lighting. In that case, I'd turn off tonemapping/sRGB on the renderer and do something similar to https://github.com/mrdoob/three.js/pull/26102 which includes tonemapping => gamma correction.

I can add a note about tone mapping and bloom to the docs, but I'd like to precede it with a solution similar to `OutputPass` linked above. Can also mention RGBA16 (`HalfFloatType`) vs SRGBA8 (`UnsignedByteType` + `SRGBColorSpace`) precision.

**Comment by vanruesc** [comment]
When using `postprocessing`, the `toneMapping` setting on the renderer should be set to `NoToneMapping` (default) because you wouldn't want colors to be mapped to [0, 1] at the start of the pipeline. If you want to use tone mapping, you can use the `ToneMappingEffect` at the end of the pipeline. This info is currently not in the readme.

> In that case, I'd turn off tonemapping/sRGB on the renderer

The `outputColorSpace` setting of the renderer is respected by `postprocessing`. There's no need for an additional pass or effect. Color space conversion is implemented as part of `EffectPass` and `CopyPass` in `postprocessing`.

Fyi: Effects also offer a way to convert between color spaces via [Effect.inputColorSpace](https://pmndrs.github.io/postprocessing/public/docs/class/src/effects/Effect.js~Effect.html#instance-get-inputColorSpace) and [Effect.outputColorSpace](https://pmndrs.github.io/postprocessing/public/docs/class/src/effects/Effect.js~Effect.html#instance-get-outputColorSpace).

> Can also mention RGBA16 (HalfFloatType) vs SRGBA8 (UnsignedByteType + SRGBColorSpace) precision.

Information about this is already included in the readme. Let me know if something should be added or changed.

**Comment by vanruesc** [comment]
Closing this as working as intended. New releases of `three` sometimes introduce breaking changes, some of which are obvious and others not so much like in this case. I think this was just a configuration issue that relied on broken behaviour.

**Comment by lviggiani** [comment]
> When using `postprocessing`, the `toneMapping` setting on the renderer should be set to `NoToneMapping` (default) because you wouldn't want colors to be mapped to [0, 1] at the start of the pipeline. If you want to use tone mapping, you can use the `ToneMappingEffect` at the end of the pipeline. This info is currently not in the readme.
> 
> > In that case, I'd turn off tonemapping/sRGB on the renderer
> 
> The `outputColorSpace` setting of the renderer is respected by `postprocessing`. There's no need for an additional pass or effect. Color space conversion is implemented as part of `EffectPass` and `CopyPass` in `postprocessing`.
> 
> Fyi: Effects also offer a way to convert between color spaces via [Effect.inputColorSpace](https://pmndrs.github.io/postprocessing/public/docs/class/src/effects/Effect.js~Effect.html#instance-get-inputColorSpace) and [Effect.outputColorSpace](https://pmndrs.github.io/postprocessing/public/docs/class/src/effects/Effect.js~Effect.html#instance-get-outputColorSpace).
> 
> > Can also mention RGBA16 (HalfFloatType) vs SRGBA8 (UnsignedByteType + SRGBColorSpace) precision.
> 
> Information about this is already included in the readme. Let me know if something should be added or changed.

Thanks for the explanation. I have a question: how do I control exposure for the `ToneMappingEffect`? Before I was using `Renderer.toneMapping = THREE.LinearTonemapping` and then changing `Renderer.toneMappingExposure` as requested. Now, since I'm not using renderer tonemapping anymore (also becasue by setting material's `.toneMapped = false` would exlude them from tonemapping and exposure too) and I wanto to use `ToneMappingEffect` instead, I still need to manually control exposure in some cases. How can this be done?

**Comment by lviggiani** [comment]
Thanks for the explanation. I'm now using a `ToneMappingEffect` (`mode: ToneMappingMode.ACES_FILMIC`) at the end of my effect pass:

- `RenderPass`
- `EccectPass: SMAAEffect, BloomEffect, ToneMappingEffect`

The godd thing is that it's keeping into consideration `Renderer.toneMappingExposure` so I can manually control it (I need it), the bad thing is that it's not rendering correctly the `Renderer.clearColor` I set. For example, I set a white color but I get a light grey instead. I also tried to move the `ToneMappingEffect` back and forth in my `EffectPass` but nothing changes.
Also the image is a bit too dark compared to `Renderer` tonemapping which I have disabled
How can I fix that?
